File: cc/app/acc-serv/internal/conf/conf.cc
//
// weproxy@foxmail.com 2022/10/03
//

#include "conf.h"
#include "conf.json.h"
#include "gx/encoding/json/json.h"

namespace internal {
namespace conf {

NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(Iface, in, out)
NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(Conf, iface, server)

// ReadConfig ...
R<json::J, error> ReadConfig() {
    return json::Parse(DEFAULT_CONF);
}

}  // namespace conf
}  // namespace internal

File: go/app/acc-serv/internal/conf/conf.go
//
// weproxy@foxmail.com 2022/10/20
//

package conf

import (
	_ "embed"
	"encoding/json"
)

//go:embed conf.json
var _DEFAULT_CONF []byte

// Conf ...
type Conf struct {
	Iface struct {
		In  string `json:"in,omitempty"`
		Out string `json:"out,omitempty"`
	} `json:"iface,omitempty"`

	Servers []json.RawMessage `json:"server,omitempty"`
}

// ReadConfig ...
func ReadConfig() (*Conf, error) {
	var c = &Conf{}
	if err := json.Unmarshal(_DEFAULT_CONF, c); err != nil {
		return nil, err
	}
	return c, nil
}
